pyerrors/input/openQCD.py

```diff
@@ -343,3 +346,241 @@ def _read_array_openQCD2(fp):
+                        raise Exception("It seems the length given in the header and by you contradict each other")
+                else:
+                    raise Exception("Found more than one spatial length in header!")
+            
+            print('dnms:', dn)
+            print('nn:', nn)
+            print('tmax:', tmax)
+            t = fp.read(8) #header4 wird mit 16 bytes ausgelesen... cmax fehlt also als Angabe
+            eps = struct.unpack('d', t)[0]# das hier ist vllt. tol
+            print('eps:', eps)
+
+            Q = []
+            ncs = []
+            while 0 < 1:
+                t = fp.read(4) #int nt
+                if(len(t) < 4):
+                    break
+                ncs.append(struct.unpack('i',t)[0])
+                t = fp.read(8 * tmax * (nn + 1))#Wsl
+                t = fp.read(8 * tmax * (nn + 1))#Ysl
+                t = fp.read(8 * tmax * (nn + 1))#Qsl, which is asked for in this method
+                #unpack the array of Qtops, on each timeslice t=0,...,tmax-1 and the 
+                #measurement number in = 0...nn (see README.qcd1)
+                tmpd = struct.unpack('d' * tmax * (nn + 1), t)
+                Q.append(tmpd)
+                #set step by reading all entries, then set stepsize, then check if everything is there
+                #make a dtr_config param, which is checked against difference...
+                #difference != step
+
+        if not len(set([ncs[i]-ncs[i-1] for i in range(1,len(ncs))])):
+            raise Exception("Irregularities in stepsize found")
+        else:
+            if 'steps' in kwargs:
+                if steps != ncs[1]-ncs[0]:
+                    raise Exception("steps and the found stepsize are not the same")
+            else:
+                steps = ncs[1]-ncs[0]
+                if ncs[0]//steps == ncs[0]/steps:
+                    r_meas_start = ncs[0]//steps
+                    
+        print(len(Q))
+        print('max_t:', dn * (nn) * eps)
+        
+        t_aim = (c * L) ** 2 / 8
+
+        print('t_aim:', t_aim)
+        index_aim = round(t_aim / eps / dn)
+        print('index_aim:', index_aim)
+
+
+        Q_sum = []
+        for i, item in enumerate(Q):
+            Q_sum.append([sum(item[current:current + tmax]) for current in range(0, len(item), tmax)])
+        print(len(Q_sum))
+        print(len(Q_sum[0]))
+        Q_round = []
+        for i in range(len(Q) // dtr_cnfg):
+            Q_round.append(round(Q_sum[dtr_cnfg * i][index_aim]))
+        if len(Q_round) != len(ncs)//dtr_cnfg:
+            raise Exception("qtops and ncs dont have the same length")
+        
+        #replica = len(files)
+
+        truncated_file = file[:-7]
+        print(truncated_file)
+        idl_start = 1
+        
+        if "r_start" in kwargs:
+            Q_round = Q_round[r_start[rep]:]
+            idl_start = r_start[rep]
+        if "r_stop" in kwargs:
+            Q_round = Q_round[:r_stop[rep]]
```

Comment:
I think the slicing here does not work properly if both `r_start` and `r_stop` are defined. I would propose something like this:
```python
if "r_start" in kwargs:
    first_cnfg = r_start[rep]
    idl_start = r_start[rep]
else:
    first_cnfg = None
    idl_start = 1
if "r_stop" in kwargs:
    last_cnfg = r_stop[rep]
else:
    last_cnfg = None

Q_round = Q_round[first_cnfg: last_cnfg]
idl_stop = idl_start+len(Q_round)
```